FWCore/FWCore/DataWrapper.h

```diff
@@ -23,8 +25,7 @@ class GAUDI_API DataWrapper : public DataWrapperBase {
   const T* getData() {return m_data;} 
   void setData(T* data) {m_data = data;}
 
-  virtual albers::CollectionBase* collectionBase(){return dynamic_cast<albers::CollectionBase*>(m_data); }
-
+  virtual albers::CollectionBase* collectionBase(){if (std::is_base_of<albers::CollectionBase,T>::value) return reinterpret_cast<albers::CollectionBase*>(m_data); return nullptr;}
```

Comment:
This line got a bit hard to read with the change. Is it possible to add a few newlines? Regarding the inlining: Not sure if this still counts as a trivial implementation :-)